File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square(n: int) -> int:
    return n * n


def square_of_sum(n: int) -> int:
    return square(sum(range(n + 1)))


def sum_of_squares(n: int) -> int:
    return sum(square(x) for x in range(n + 1))


def difference_of_squares(n: int) -> int:
    return square_of_sum(n) - sum_of_squares(n)

File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number)
{
    unsigned s = 0;
    for (unsigned x = 1; x <= number; ++x)
    {
        s += x * x;
    }
    return s;
}

unsigned int square_of_sum(unsigned int number)
{
    unsigned s = 0;
    for (unsigned x = 1; x <= number; ++x)
    {
        s += x;
    }
    return s * s;
}

unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}
